hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -229,7 +229,7 @@ public class HoodieTableConfig extends HoodieConfig {
 
   public static final ConfigProperty<HoodieTimelineTimeZone> TIMELINE_TIMEZONE = ConfigProperty
       .key("hoodie.table.timeline.timezone")
-      .defaultValue(HoodieTimelineTimeZone.LOCAL)
+      .defaultValue(HoodieTimelineTimeZone.UTC)
```

Comment:
https://github.com/apache/hudi/pull/12926#discussion_r1999392569

Reply:
Consider always setting the timezone in the table config if we're flipping the default.